packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/PersistedNameTests.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.test.compiler
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import io.realm.kotlin.compiler.Registrar
+import io.realm.kotlin.test.util.Compiler
+import org.junit.Test
+import kotlin.test.assertContains
+import kotlin.test.assertEquals
+
+class PersistedNameTests {
+    @Test
+    fun `empty annotation fails`() {
+        val result = Compiler.compileFromSource(
+            plugins = listOf(Registrar()),
+            source = SourceFile.kotlin(
+                "persistedNameAnnotationEmpty.kt",
+                """
+                    import io.realm.kotlin.types.RealmObject
+                    import io.realm.kotlin.types.annotations.PersistedName
+                    
+                    class InvalidSample : RealmObject {
+                        @PersistedName("")
+                        var publicName1: String? = ""
+                    }
+                """.trimIndent()
+            )
+        )
+
+        assertEquals(KotlinCompilation.ExitCode.COMPILATION_ERROR, result.exitCode)
+        val compilerLog = result.messages
+        assertContains(compilerLog, "Names must contain at least 1 character")
+    }
+
+    @Test
+    fun `same persisted and public name warns`() {
+        val result = Compiler.compileFromSource(
+            plugins = listOf(Registrar()),
+            source = SourceFile.kotlin(
+                "samePersistedAndPublicName.kt",
+                """
+                    import io.realm.kotlin.types.RealmObject
+                    import io.realm.kotlin.types.annotations.PersistedName
+                    
+                    class InvalidSample : RealmObject {
+                        @PersistedName("sameName")
+                        var sameName: String? = ""
+                    }
+                """.trimIndent()
+            )
+        )
+
+        assertEquals(KotlinCompilation.ExitCode.OK, result.exitCode)
+        val compilerLog = result.messages
+        assertContains(compilerLog, "The Kotlin name and the persisted name are the same value: 'sameName'")
```

Comment:
Nitpick, throwing a build error on this is a little pedantic since it would be valid, just redundant. Can we turn this into a warning instead?

Reply:
Would this be a build error? Wouldn't it be an assertion error at runtime?

Reply:
Hmm, maybe I misread this. I can see that the return value is `KotlinCompilation.ExitCode.OK` which seems to indicate that it works correctly.

Reply:
Yes this is only a warning (not an error) via `logDebug()`.